app/models/concerns/stock_settings_override_validation.rb

```diff
@@ -1,3 +1,29 @@
+# Validates the combination of on_demand and count_on_hand values.
+#
+# `on_demand` can have three values:
+#
+# - true
+# - false
+# - nil
+#
+# `count_on_hand` can either be:
+#
+#  - nil
+#  - a number
+#
```

Comment:
My preference goes for using less lines in comments so I'd do:
```suggestion
# `on_demand` can have three values: true, false or nil
# `count_on_hand` can either be: nil or a number 
```

Reply:
Good idea. I committed your changes.